docs/Development-Guides/Errors/Error-Types.md

```diff
@@ -35,4 +35,5 @@ Name | Finch Code | Code | Description
 `rate_limit_exceeded_error` | `finch_rate_limit_exceeded` | 429 | The application has exceeded Finch's rate limits. Please retry later.
 `rate_limit_exceeded_error` | `upstream_rate_limit_exceeded` | 429 | The application has exceeded upstream provider rate limits. Please retry later.
 `server_error` | | 500 | The server experienced an unexpected error.
+`bad_gateway_error` | `upstream_provider_error` | 502 | The server experienced an unexpected error while interacting with an upstream service, such as a provider.
```

Comment:
our current naming schema looks like
name: http status name
finch_name: more specific name relating the error to finch

current proposal here is in line with this. what do we think?
would require updating the api-server PR to match this

Reply:
LGTM. Let's wait on Shoabe to weigh in.
I'll wait for this to be merged to change the PR with the finch_code, in case it changes before approve and merge.

Reply:
@eliseuvideira1 let's make the changes on the api-server side and merge, and then i will merge this PR while the api-server PR is deploying